lib/tests/assertions.rb

```diff
@@ -9,75 +9,122 @@ def assert(test, message="assertion failed, no message", data="")
+        unless assertion_bool( response.resource.class == FHIR::Bundle )
           raise AssertionException.new "Expected FHIR Bundle but found: #{response.resource.class}", response.body
         end
       end
+      def assert_bundle_entry_count(response, count)
+        unless assertion_bool( response.resource.total == count.to_i )
+          raise AssertionException.new "Expected FHIR Bundle with #{count} entries but found: #{response.resource.total} entries", response.body
+        end
+      end
 
       def assert_valid_resource_content_type_present(client_reply)
         header = client_reply.response.headers[:content_type]
-        
+
         content_type = header
         charset = encoding = nil
-        
+
         content_type = header[0, header.index(';')] if !header.index(';').nil?
         charset = header[header.index(';charset=')+9..-1] if !header.index(';charset=').nil?
         encoding = Encoding.find(charset) if !charset.nil?
-        
-        unless encoding == Encoding::UTF_8
+
+        unless assertion_bool( encoding == Encoding::UTF_8 )
           raise AssertionException.new "Response content-type specifies encoding other than UTF-8: #{charset}", header
         end
-        unless (content_type == FHIR::Formats::ResourceFormat::RESOURCE_XML) || (content_type == FHIR::Formats::ResourceFormat::RESOURCE_JSON)
+        unless assertion_bool( (content_type == FHIR::Formats::ResourceFormat::RESOURCE_XML) || (content_type == FHIR::Formats::ResourceFormat::RESOURCE_JSON) )
           raise AssertionException.new "Invalid FHIR content-type: #{content_type}", header
         end
       end
 
       def assert_last_modified_present(client_reply)
         header = client_reply.response.headers[:last_modified]
-        assert !header.nil?, 'Last-modified HTTP header is missing.'
-      end  
-      
+        assert assertion_bool( !header.nil? ), 'Last-modified HTTP header is missing.'
+      end
+
       def assert_valid_content_location_present(client_reply)
         header = client_reply.response.headers[:content_location]
-        assert !header.nil?, 'Content-location HTTP header is missing.'
-      end   
+        assert assertion_bool( !header.nil? ), 'Content-location HTTP header is missing.'
+      end
+
+      def assert_response_code(response, code)
+        unless assertion_bool( code == response.code )
+          raise AssertionException.new "Bad response code expected #{code}, but found: #{response.code}", response.body
+        end
+      end
+
+      def assert_resource_type(response, resource_type)
+        unless assertion_bool( !response.resource.nil? && response.resource.class == resource_type )
+          raise AssertionException.new "Bad response type expected #{resource_type}, but found: #{response.resource.class}", response.body
+        end
+      end
+
+      def assert_minimum(response, fixture)
+        resource_xml = response.try(:resource).try(:to_xml)
```

Comment:
Should be compared against the raw response, not the deserialized and reserialized XML.